### ISSUE TITLE
Disambiguation dialog is shown when opening activity with URI

### ISSUE THREAD

**Issue by nhkarthick** [googlesamples/android-dynamic-features#55]
The project doesn't build in Android Studio version 3.4.1. So I downgraded the gradle version to 3.4.2 (com.android.tools.build:gradle:3.4.2). 

**Steps to repro**
1. Click **REQUEST INSTALL SPLIT INSTALL VIA URL** 
2. An disambiguation dialog is shown with apps of the same package name. 

Looking forward for help. Thanks

**Comment by Entreco** [comment]
I have the same problem, and created a simple repository to mimic my setup. Off course, when trying out this project, make sure you update the domain and packagename of the app, to reflect your assetlinks.json file

https://github.com/Entreco/Dynamic-feature-disambiguation/tree/master

**Comment by Karthi-R** [comment]
Found an open issue [here](https://issuetracker.google.com/issues/123857941) in issue tracker. Our entire application depends on this URI intent navigation. This bug is complete show stopper on implementing dynamic feature.

@keyboardsurfer do you have any update on this fix? Any workaround do you suggest ?

Appreciate your help!

**Comment by Entreco** [comment]
I asked the same question on [StackOverflow](https://stackoverflow.com/questions/57340444/instant-app-with-dynamic-features-always-show-disambiguation-dialog-with-1-optio). 

Until the issue is fixed, the suggested work-around is to:

> So yes... I believe I have seen this before, it is some odd behavior when navigating from one dynamic-feature (instant) to another (non-instant) via a URL intent.
>
>Until this gets addressed, I don't recommend using a URL intent to navigate between modules, instead, use reflection to directly get to the other module's activity, example:
> 
> if (doesModuleExist()) {
>     val intent = Intent()
>         .setClassName(getPackageName(), "com.sample.ProfileActivity")
>         .addCategory(Intent.CATEGORY_DEFAULT)
>         .addCategory(Intent.CATEGORY_BROWSABLE)
>     startActivity(intent)
> }

Which basically means, to use URL intents only for **_starting_** your instant app/dynamic features. But navigation **_between_** features in your app, should use reflection

**Comment by Karthi-R** [comment]
@Entreco Thanks for your update!

we believe, using reflection is not a scalable solution. also it involves lot of code changes. 
Any quick workaround available on this ?

**Comment by Entreco** [comment]
Agreed that reflection feels wrong/ awkward. But that's the only work around I have heard so far. 

Would be nice to get an update about this issue from people working on the project

**Comment by keyboardsurfer** [comment]
In order to launch an activity via URL without the disambiguation dialog you might have to call [`setPackage`](https://developer.android.com/reference/android/content/Intent.html#setPackage(java.lang.String)) with your package name to handle the intent.

**Comment by Entreco** [comment]
I see the disambiguation dialog in my sample project: https://github.com/Entreco/Dynamic-feature-disambiguation

Calling `setPackage` does not seem to make a difference (at least with my setup, device and buildtools ):
```
val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://giddy.entreco.nl/viewer/"))
            .setPackage(packageName)
            .addCategory(Intent.CATEGORY_DEFAULT)
            .addCategory(Intent.CATEGORY_BROWSABLE)
        startActivity(intent)
```

**Comment by Karthi-R** [comment]
Thanks @keyboardsurfer for the quick response.
As @Entreco mentioned; Calling `setPackage` doesn't fix this issue.
